src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3346,6 +3358,176 @@ public AttributeProviderPair (CustomAttribute attribute, ICustomAttributeProvide
+										int argsParam = parameters.Count == 2 || parameters.Count == 3 ? 1 : 3;
+										
+										if (methodParams.Count > argsParam &&
+											methodParams [argsParam] is ArrayValue arrayValue &&
+											arrayValue.Size.AsConstInt () != null) {
+											ctorParameterCount = arrayValue.Size.AsConstInt ();
+										}
+
+										if (parameters.Count > 3) {
+											if (methodParams [1].AsConstInt () != null)
+												bindingFlags |= (BindingFlags)methodParams [1].AsConstInt ();
+											else
+												bindingFlags |= BindingFlags.NonPublic | BindingFlags.Public;
+										} else {
+											bindingFlags |= BindingFlags.Public;
+										}
+									}
+								}
+								else {
+									// The overload with a single System.Type argument
+									ctorParameterCount = 0;
+									bindingFlags |= BindingFlags.Public;
+								}
+
+								// Go over all types we've seen
+								foreach (var value in methodParams[0].UniqueValues ()) {
+									if (value is SystemTypeValue systemTypeValue) {
+										MarkMethodsFromReflectionCall (ref reflectionContext, systemTypeValue.TypeRepresented, ".ctor", bindingFlags, ctorParameterCount);
+									} else if (value == NullValue.Instance) {
+										// Nothing to report. This is likely just a value on some unreachable branch.
+										reflectionContext.RecordHandledPattern ();
+									} else if (value is MethodParameterValue methodParameterValue) {
+										// This is the case where the value comes from a method parameter.
+										// TODO: If the parameter is annotated, we're good. If it's not annotated, we shold warn.
+										reflectionContext.RecordUnrecognizedPattern ($"Activator call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 1st argument expression which cannot be analyzed");
+									} else {
+										// Not known where the value is coming from
+										reflectionContext.RecordUnrecognizedPattern ($"Activator call '{calledMethod.FullName}' inside '{callingMethodBody.Method.FullName}' was detected with 1st argument expression which cannot be analyzed");
+									}
+								}
+							}
+							break;
+						default:
+							return false;
+					}
+				}
+				finally {
+					reflectionContext.Dispose ();
+				}
+
+				// If we get here, we handled this as an intrinsic.  As a convenience, if the code above
+				// didn't set the return value (and the method has a return value), we will set it to be an
+				// unknown value with the return type of the method.
+				if (methodReturnValue == null) {
+					if (calledMethod.ReturnType.MetadataType != MetadataType.Void) {
+						methodReturnValue = UnknownValue.Instance;
+					}
+				}
+
+				return true;
+			}
+
+			void MarkMethodsFromReflectionCall (ref ReflectionPatternContext reflectionContext, TypeDefinition declaringType, string name, BindingFlags? bindingFlags, int? parametersCount = null)
```

Comment:
This is copypaste of the existing method in this file. I removed the generic arity parameter because generic methods don't have mangled names. I don't think that code path is ever reachable. I expect when we're done, the existing method will be deleted in favor of this scanner, so I didn't bother unifying.